### ISSUE TITLE
🐛 BUG: Empty HTML tag (fragment) in root of astro file leads to crash of language server in VSCode

### ISSUE THREAD

**Issue by beeb** [withastro/language-tools#518]
### Describe the Bug

```
Successfully initialized! 🚀
panic: runtime error: slice bounds out of range [-1:]

goroutine 6 [running]:
github.com/withastro/compiler/internal/printer.renderTsx(0x641a98, 0x692640)
	github.com/withastro/compiler/internal/printer/print-to-tsx.go:427 +0x5a8
github.com/withastro/compiler/internal/printer.renderTsx(0x641a98, 0x6923c0)
	github.com/withastro/compiler/internal/printer/print-to-tsx.go:134 +0xdb
github.com/withastro/compiler/internal/printer.PrintToTSX({0x4867c8, 0x16}, 0x6923c0, {{0x3913f, 0x6}, {0x4af860, 0x4d}, {0x4af860, 0x4d}, {0x438d4, ...}, ...}, ...)
	github.com/withastro/compiler/internal/printer/print-to-tsx.go:25 +0xc
main.ConvertToTSX.func1({{}, 0x7ff800010000000d, 0x6255c8}, {0x42ca80, 0x2, 0x2})
	./astro-wasm.go:237 +0x2a
syscall/js.handleEvent()
	syscall/js/func.go:94 +0x26
TypeError: Cannot read properties of undefined (reading 'map')
    at file:///home/valentin/.vscode-server/extensions/astro-build.astro-vscode-1.0.0-linux-x64/node_modules/@astrojs/compiler/node/index.js:61:60
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: Go program has already exited
    at Go._resume (file:///home/valentin/.vscode-server/extensions/astro-build.astro-vscode-1.0.0-linux-x64/node_modules/@astrojs/compiler/node/wasm_exec.js:435:19)
    at Object.convertToTSX (file:///home/valentin/.vscode-server/extensions/astro-build.astro-vscode-1.0.0-linux-x64/node_modules/@astrojs/compiler/node/wasm_exec.js:447:16)
    at Promise.then (file:///home/valentin/.vscode-server/extensions/astro-build.astro-vscode-1.0.0-linux-x64/node_modules/@astrojs/compiler/node/index.js:60:62)
    at new Promise (<anonymous>)
    at Object.convertToTSX (file:///home/valentin/.vscode-server/extensions/astro-build.astro-vscode-1.0.0-linux-x64/node_modules/@astrojs/compiler/node/index.js:60:20)
    at file:///home/valentin/.vscode-server/extensions/astro-build.astro-vscode-1.0.0-linux-x64/node_modules/@astrojs/compiler/node/index.js:11:51
[Info  - 11:05:11 AM] Connection to server got closed. Server will restart.
```

### Steps to Reproduce

1. Open Astro project in VSCode
2. Create `.astro` component or page and insert the following `<>foobar</>`
3. Save file
4. "The Astro server crashesd 5 times in the last 3 minutes. The server will not be restarted. See the output for more information."

**Comment by Princesseuh** [comment]
One of your Astro file is crashing the compiler. It's unfortunate that you cannot share the repo, but if you manage to isolate the file that does it, that'd be awesome

**Comment by beeb** [comment]
@Princesseuh is it possible that the compiler crashes on one of my files when the actual build runs fine?
There are hundreds of files in the project, I would love to be able to narrow it down somewhat.

**Comment by beeb** [comment]
Got lucky and found the culprit:

```astro
// PhoneNumber.astro
<>+0123456789</>
```

If I remove the "fragment" empty tag it works fine.

**Comment by Princesseuh** [comment]
Ah, awesome! We have tests for this, so weird that it's crashing, but I'll take a look. Thank you!

In the meantime, you can replace it with `<Fragment>+0123456789</Fragment>` for the same result

**Comment by beeb** [comment]
Thanks for the hint, I actually just removed the tag completely and the result seems to be the same, I think I don't need the fragment here.

**Comment by Princesseuh** [comment]
> @beeb I've just been dealing with a similar (the same?) issue with the astro language tools crashing on vscode. It turns out I was able reproduce this by introducing an astro component that contains a non-standard line terminator: [symbl.cc/en/2028](https://symbl.cc/en/2028/)
> 
> The reason I was having an issue was because I stupidly left behind a local untracked file containing of of these; which explains why none of my coworkers were getting the same problem!
> 
> If you're stuggling to find the offending file and you really, really love horribly hacky solutions, you might have some luck throwing some console logs into `.vscode-server/extensions/astro-build.astro-vscode-1.0.0-linux-x64/node_modules/@astrojs/compiler/node/index.js`. The `options` param in
> 
> ```
>         convertToTSX: (input, options) => {
> ```
> 
> contains the name of the file being converted, so it's easy enough to see which file it's getting stuck on

Thank you, filed an issue upstream https://github.com/withastro/compiler/issues/785

Definitely a edge case, as using one of those character in a file will pop-up a warning in a lot of editors, but still, we shouldn't crash for sure.

**Comment by Princesseuh** [comment]
Today's update of the extension adds the filename that failed to compile and prevent the extension from crashing when it happens. Hopefully that make things smoother until all the crashing issues are fixed!

**Comment by Princesseuh** [comment]
This has been fixed as of `2.1.0`!
